src/app/core/services/logging.service.ts

```diff
@@ -7,28 +7,106 @@ import { ElectronLog } from 'electron-log';
 })
 export class LoggingService {
   private logger: ElectronLog | Console;
+  private readonly LOG_KEY = 'CATcher-Log';
+  private readonly LOG_FILE_NAME = 'CATcher-log.txt';
+  private readonly LOG_START_HEADER = '====== Initializing CATcher Log ======';
+  private readonly LOG_COUNT_LIMIT = 4;
 
   constructor(electronService: ElectronService) {
     if (electronService.isElectron()) {
       this.logger = window.require('electron-log');
     } else {
       this.logger = console;
     }
+
+    this.initializeLogCache();
+  }
+
+  initializeLogCache() {
+    this.setCachedLog(this.getTrimmedLogCache(this.getCachedLog(), this.LOG_COUNT_LIMIT));
+  }
+
+  /**
+   * Trims the existing Log in the browser's cache to a select number
+   * of Sessions if necessary.
+   * @param sessionCount The number of Session Logs to preserve in the cache
+   */
+  getTrimmedLogCache(currentLog: string, sessionCount: number): string {
+    // Check if Trimming is Necessary
+    const numberOfSessions: number = currentLog == null ? 0 : currentLog.split('\n')
+      .filter((currentLogLine: string) => currentLogLine.includes(this.LOG_START_HEADER))
+      .length;
+
+    if (numberOfSessions < sessionCount) {
+      return `${currentLog}\n${this.LOG_START_HEADER}`;
+    }
+
+    const seperatedSessionLogs: string[] = [''];
+    currentLog.split('\n')
+      .forEach((currentLogLine: string) => {
+        if (currentLogLine === this.LOG_START_HEADER) {
+          seperatedSessionLogs.push(currentLogLine);
+        } else {
+          seperatedSessionLogs[seperatedSessionLogs.length - 1] += `\n${currentLogLine}`;
+        }
+      });
```

Comment:
```suggestion
    let separatedSessionLogs = currentLog.split(this.LOG_START_HEADER)
      .filter(line => !!line) // remove first empty line
      .map(line => this.LOG_START_HEADER + line); // add header back
```

I think we can use the `let` declaration since `separatedSessionLogs` is going to be modified in-place.

Reply:
Ah yes this is a good strategy. I think just need to account for the edge case where `currentLog` is `null` and this would be cleaner. Thanks for the suggestion 👍

Reply:
Ahh yes this would be cleaner. Thanks for the suggestion 👍